pkg/server/cloud/aws/instances.go

```diff
@@ -359,6 +359,130 @@ func (e *AwsEC2) StartNode(node *api.Node, image cloud.Image, metadata, iamPermi
 	return cloudID, nil
 }
 
+// We need to ensure the dedicated host fulfills two constraints
+// 1) in a state of "available"
+// 2) no tenant is currently occupying the host
+func (e *AwsEC2) ReleaseDedicatedHosts() error {
+	hosts, err := e.listAvailableDedicatedHosts()
+	if err != nil {
+		return err
+	}
+	var hostIdsForRelease []*string
+	for _, host := range hosts {
+		if len(host.Instances) > 0 {
+			continue
+		}
+		hostIdsForRelease = append(hostIdsForRelease, host.HostId)
+	}
+	// TODO do we need to know which hosts were successfully released vs which
+	// were unsuccessful? this runs within the gcLoop so it should clean up
+	// anything not cleaned on a next iteration
+	_, err = e.client.ReleaseHosts(&ec2.ReleaseHostsInput{
```

Comment:
I think we should log about unsuccessful release. Otherwise if the releasing fails and keep failing, it would be hard to debug

Reply:
I have made updates to show the host id as well as the associated error if a host is unsuccessfully released. I have also added a comment stating why we should not be returning these errors